### PR TITLE
Specific apps feature added

### PR DESCRIPTION
**Pull request by AmniX**
Hi, I have added feature to select packages so user can Intercept selected packages only.

I required it for myself, But I would like to contribute this too.

Some Screenshots are below.

image
image

### CODE REVIEW

**Comment by AmniX** [comment]
Travis CI Failed Because of Below Error.
The command "openssl aes-256-cbc -K $encrypted_410c058128c0_key -iv $encrypted_410c058128c0_iv -in android-travis-cert-store.enc -out ./android-travis-cert-store.jks -d" failed and exited with 1 during .

**Comment by pimterry** [comment]
Amazing! This looks really good, great work. Everything's a little hectic now so it's going to take me a couple of days to get time to properly dig into this and test it out, but it looks pretty solid from a quick skim.

Don't worry about the Travis error. It's complaining because PRs can't read the env var secrets used on the master build. I should tidy that up really, but for now you can just ignore it.

Have you seen the [Free As In Beer](https://httptoolkit.tech/blog/free-as-in-beer#new-decade-resolutions) policy? In short, contributors get free HTTP Toolkit Pro. I'll sort that out once I've reviewed this and merged it, but do let me know what email address you want your account linked to once this is done - just send me an email to tim@httptoolkit.tech.

File: app/src/main/java/tech/httptoolkit/android/ApplicationListActivity.kt [hunk 1]
```diff
@@ -0,0 +1,149 @@
...
+        }
+    }
+
+    companion object {
+        fun getWhiteListAppSharedPreferences(ctx: Context): SharedPreferences =
+            ctx.getSharedPreferences("whiteListedApps", Context.MODE_PRIVATE)
+    }
+
+    override fun onBackPressed() {
+        Toast.makeText(this, "Disconnect and Connect again to take effect.", Toast.LENGTH_LONG)
```
[the remark this answers is not in the document]
Comment (AmniX) [reply app/src/main/java/tech/httptoolkit/android/ApplicationListActivity.kt:145]: Yes, We can Implement auto reconnect easily.

File: app/src/main/java/tech/httptoolkit/android/ProxyVpnService.kt [hunk 2]
```diff
@@ -166,15 +162,8 @@ class ProxyVpnService : VpnService(), IProtectSocket {
...
-                    // For some reason, with Genymotion the whole device crashes if we intercept
-                    // blindly, but intercepting every single application explicitly is fine.
-                    if (isGenymotion) {
-                        packageNames.forEach { name ->
-                            if (name != httpToolkitPackage) addAllowedApplication(name)
-                        }
-                    } else {
-                        addDisallowedApplication(httpToolkitPackage)
+                    packageNames.forEach { name ->
+                        if (name != httpToolkitPackage) addAllowedApplication(name)
```
[the remark this answers is not in the document]
Comment (AmniX) [reply app/src/main/java/tech/httptoolkit/android/ProxyVpnService.kt:166]: Hmm.
Nice Point.

Will take care of this while doing suggested changes.

File: app/src/main/res/layout/item_app_row.xml [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+            android:layout_height="wrap_content"
+            android:layout_below="@id/row_app_name"
+            android:layout_toStartOf="@id/row_app_check_box"
+            android:layout_toEndOf="@id/row_app_icon_image"
+            android:maxLines="2"
+            android:paddingStart="16dp"
+            android:paddingEnd="16dp"
+            tools:text="@tools:sample/lorem/random" />
+
+    <com.google.android.material.checkbox.MaterialCheckBox
```
[the remark this answers is not in the document]
Comment (AmniX) [reply app/src/main/res/layout/item_app_row.xml:42]: You Mean Switch Button or Toggle ON OFF Button?

**Comment by AmniX** [comment]
Thank you so much for In Depth Code Review and Suggestions.

Will Implement over the weekend.

File: app/src/main/res/layout/item_app_row.xml (hunk 3 above)
[the remark this answers is not in the document]
Comment (pimterry) [reply app/src/main/res/layout/item_app_row.xml:42]: You're right, I mean a switch: https://material.io/develop/android/components/switches/

**Comment by pimterry** [comment]
Hi @AmniX, any update on this? No problem if you don't have time, I can dig into it myself, just let me know.

**Comment by AmniX** [comment]
Yes Please Go ahead. Busy in some office work which is super big that i even working on weekends.

**Comment by pimterry** [comment]
I'm just getting around to finishing this myself now, so I'm just rebasing it and handling my comments now, and then I'll merge this shortly.

@AmniX, sorry for the delay, and thanks again for this contribution! Are you still interested in the free Pro account? Send me an email to tim@httptoolkit.tech and I can get set you up straight away if so.

I've also activated the licensing check on this repo, that should've been here before - when you get a sec, can you go to https://cla-assistant.io/httptoolkit/httptoolkit-android?pullRequest=1, log in with your github account, and accept the agreement there? Just takes two clicks, it's just a matter of confirming that you do have the rights to contribute this code, and agreeing to give the rights to use this contribution to the HTTP Toolkit project. The challenges of an open-source business :smile:

**Pull request merged**